### PR TITLE
Integrate the HiFi 5 Neural Network Library v1.5.0

### PR DESCRIPTION
**Pull request by bhanuprakashbv**
Integrated HiFi 5 NN Library kernels for below operators
  svdf(int8),
  quantize(int8),
  add(int8, uint8 and float32),
  mul(int8, uint8 and float32),
  relu(int8, uint8 and float32) and
  relu6(int8, uint8 and float32)

Signed-off-by: Anirban Mandal <amandal@cadence.com>
Signed-off-by: Biswa Dhal <bdhal@cadence.com>
Signed-off-by: Bhanu Prakash Bandaru Venkata <bhanup@cadence.com>
Signed-off-by: Chaitanya Sanjay Muley <cmuley@cadence.com>
Signed-off-by: Harinarayanan E V <hariev@cadence.com>
Signed-off-by: Harshavardhan Ravindra Joshi <joshih@cadence.com>

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/quantize.cc [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+******************************************************************************/
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/xtensa/quantize.cc:21]: Similar to the other already merged Xtensa kernels, we will need to remove this copyright. The only permissible copyright is for the Tensorflow Authors.

File: tensorflow/lite/micro/kernels/xtensa/svdf.cc [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+******************************************************************************/
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/xtensa/svdf.cc:21]: remove

File: tensorflow/lite/micro/kernels/xtensa/xtensa.h [hunk 3]
```diff
@@ -1,3 +1,24 @@
...
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/xtensa/xtensa.h:20]: remove

File: tensorflow/lite/micro/kernels/xtensa/xtensa.h [hunk 4]
```diff
@@ -22,6 +43,12 @@ limitations under the License.
 #include "include/nnlib/xa_nnlib_api.h"
 #include "include/nnlib/xa_nnlib_standards.h"
 
+#define CHECK_ERR_HIFI_NNLIB_KER(ret, err_msg) \
+  if (ret != 0) {                              \
+    TF_LITE_KERNEL_LOG(context, err_msg);      \
+    return kTfLiteError;                       \
+  }
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/xtensa/xtensa.h:50]: let's directly use the tflite macros (TF_LITE_EXPECT_EQ(context, ..., 0);

**Review by advaitjain — CHANGES_REQUESTED** [review]
For mul, add and activations we will need to first refactor the reference kernels so that the code that is shared between the reference and optimized implementations is reused -- let's tackle these ops in separate PRs.

**Comment by bhanuprakashbv** [comment]
@advaitjain, updated the code based on your review comments (commit id:725775df1d10713e91d356cdf470a18d50e2d40a).
Currently removed add.cc, mul.cc and activations.cc from this PR. Let me know when can I integrate for these operators.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**